### PR TITLE
Fixed editor freezing when providing non-integer colspan or rowspan attributes values

### LINKED ISSUE #10042: Source editing can freeze the editor when pasting some table html
## 📝 Provide detailed reproduction steps (if any)

1. Open all-features manual test.
2. Open source editing mode.
3. Replace the current source with the one from the snippet below.
4. Go back to editable.

```html
<figure class=\"table\" style=\"width:597px;\"><table><thead><tr><th style=\"background-color:hsl(0, 0%, 60%);\" colspan=\"2\"><span style=\"color:hsl(0,0%,100%);\">Saturday, July 14</span></th></tr></thead><tbody><tr><td style=\"background-color:hsl(0, 0%, 90%);text-align:center;\">9:30 AM - 11:30 AM</td><td><p><strong>Americano vs. Brewed - “know your coffee”</strong> with:&nbsp;</p><ul><li>Giulia Bianchi</li><li>Stefano Garau</li><li>Giuseppe Russo</li></ul></td></tr><tr><td style=\"background-color:hsl(0, 0%, 90%);text-align:center;\">1:00 PM - 3:00 PM</td><td><p><strong>Pappardelle al pomodoro</strong> - live cooking</p><p>Incorporate the freshest ingredients&nbsp;<br>with Rita Fresco</p></td></tr><tr><td style=\"background-color:hsl(0, 0%, 90%);text-align:center;\">5:00 PM - 8:00 PM</td><td><strong>Tuscan vineyards at a glance</strong> - wine-tasting&nbsp;<br>with Frederico Riscoli</td></tr></tbody></table></figure><p>&nbsp;</p>
```

### ✔️ Expected result

The editor opens with new content set during source editing.

### ❌ Actual result

Editor freezes, browser tab is unresponsive. There are no errors in the console.

## 📃 Other details

* Browser: Chrome
* OS: macOS BigSur 11.4
* Content from the above snippet was taken from Export-to-Pdf demo from our docs. It was escaped by default in Chrome dev tools console.
* If this string will be unescaped before pasting to source editing then it works correctly
* It looks like the issue is related to this attribute: `colspan=\"2\"`, changing it to `colspan="2"` without changing anything else will correctly open new content in the editor

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by dawidurbanski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (table): Fixed editor freezing when providing non-integer colspan or rowspan attributes values. Closes #10042.

---

### Additional information

Can be tested by running `yarn manual --files=table` and checking `tickets/10042.html` test case.

### CODE REVIEW

**Comment by dawidurbanski** [comment]
@niegowski I could also try to remove `( type === 'colspan' && span <= 0 ) || ( type === 'rowspan' && span < 0 )` check and leave it to HTML parser.

File: packages/ckeditor5-table/src/tableediting.js [hunk 1]
```diff
@@ -113,9 +113,29 @@ export default class TableEditing extends Plugin {
 			converterPriority: 'high'
 		} );
 
+		// Fixes invalid colspan and rowspan attrbutes values
+		const upcastCellSpan = type => {
```
Comment (niegowski) [inline packages/ckeditor5-table/src/tableediting.js:117]: This should be extracted to a local function in this module.

File: packages/ckeditor5-table/src/tableediting.js [hunk 2]
```diff
@@ -113,9 +113,29 @@ export default class TableEditing extends Plugin {
...
+			return cell => {
+				const span = parseInt( cell.getAttribute( type ) );
+
+				if ( Number.isNaN( span ) || ( type === 'colspan' && span <= 0 ) || ( type === 'rowspan' && span < 0 ) ) {
```
Comment (niegowski) [inline packages/ckeditor5-table/src/tableediting.js:121]:
```suggestion
				if ( Number.isNaN( span ) || span <= 0 ) {
```

File: packages/ckeditor5-table/src/tableediting.js [hunk 3]
```diff
@@ -113,9 +113,29 @@ export default class TableEditing extends Plugin {
...
+					return 1;
```
Comment (niegowski) [inline packages/ckeditor5-table/src/tableediting.js:122]:
```suggestion
					return null;
```

File: packages/ckeditor5-table/src/tableediting.js [hunk 4]
```diff
@@ -113,9 +113,29 @@ export default class TableEditing extends Plugin {
...
+			};
+		};
+
 		// Table attributes conversion.
-		conversion.attributeToAttribute( { model: 'colspan', view: 'colspan' } );
-		conversion.attributeToAttribute( { model: 'rowspan', view: 'rowspan' } );
+		conversion.for( 'downcast' ).attributeToAttribute( { model: 'colspan', view: 'colspan' } );
+		conversion.for( 'upcast' ).attributeToAttribute(
+			{ model: { key: 'colspan', value: upcastCellSpan( 'colspan' ) }, view: 'colspan' }
+		);
```
Comment (niegowski) [inline packages/ckeditor5-table/src/tableediting.js:133]:
```suggestion
		conversion.for( 'upcast' ).attributeToAttribute( { 
			model: { key: 'colspan', value: upcastCellSpan( 'colspan' ) }, 
			view: 'colspan' 
		} );
```

**Review by niegowski — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-table/src/tableediting.js [hunk 5]
```diff
@@ -159,3 +168,20 @@ export default class TableEditing extends Plugin {
 		return [ TableUtils ];
 	}
 }
+
+// Returns fixed colspan and rowspan attrbutes values
```
Comment (niegowski) [inline packages/ckeditor5-table/src/tableediting.js:172]:
```suggestion
// Returns fixed colspan and rowspan attrbutes values.
```

**Pull request merged**